libcudacxx/test/libcudacxx/std/utilities/optional/optional.comp_with_t/equal.pass.cpp

```diff
@@ -34,6 +34,20 @@ __host__ __device__ constexpr bool operator==(const X& lhs, const X& rhs)
   return lhs.i_ == rhs.i_;
 }
 
+struct DeviceOnly
+{
+  __device__ friend constexpr bool operator==(const DeviceOnly& lhs, const DeviceOnly& rhs)
+  {
+    return true;
+  }
+};
+
+__global__ void ensure_exec_check_is_disabled()
```

Comment:
question: Do we need to define a separate `__global__` function? The `fake_main()` thing already makes `main()` a `__global__` function, right?

Can't we do something like:
```
int main(){
   NV_IF_TARGET(NV_IS_DEVICE, assert(optional<DeviceOnly>{} == DeviceOnly{}));
}
```
? 
}

Reply:
I do not really understand why, but that is what I initially did and it did not emit any errors.

Probably the same that using global objects like `cuda::std::nullopt` only triggers errors in a kernel

Reply:
Could be because `main()` actually expands to a `__host__ __device__` function which is then invoked from a `__global__`. That could interfere with if/when nvcc emits a diagnostic. 

https://github.com/NVIDIA/cccl/blob/e3d181f0a3c8d2f49aec72e04d26b9b386f92b52/libcudacxx/test/libcudacxx/force_include.h#L49-L57